docs/libcudacxx/extended_api/warp/warp_shuffle.rst

```diff
@@ -125,7 +125,7 @@ The functions allow to exchange data of any data size, including raw arrays, poi
 **Constrains**
 
 - ``Width`` must be a power of two in the range [1, 32]
-- ``T``: only ``void`` pointers are allowed to avoid bug-prone code
+- ``T``: all ``T`` are allowed except if ``T`` is a pointer, in which case it must be a ``void`` pointer to avoid buggy programs.
```

Comment:
It'd be good to specify which sort of bugs this avoids.
I don't see any significant issue with allowing any pointers (*), so calling those out would have been helpful.

(*) Particularly since any type containing pointers - like smart pointers/references, iterators, ... - are allowed.